regression_utils.py

```diff
@@ -0,0 +1,45 @@
+import numpy as np
+
+def reject_outliers(features_table: np.ndarray[np.ndarray], values_to_be_predicted: list, upper_percentile: int, lower_percentile: int, outlierConstant: float) -> (list[list], list):
```

Comment:
not sure if the `values_to_be_predicted` type is `list`
```suggestion
def reject_outliers(features_table: np.ndarray[np.ndarray], values_to_be_predicted: list, upper_percentile: int, lower_percentile: int, outlierConstant: float) -> (np.ndarray[np.ndarray], np.ndarray):
```
Though type hinting of `numpy` structures is tricky. Numpy has a special type hinting module with an array typing:
```
from numpy.typing import NDArray
``` 
not super useful lolp ([discussion](https://stackoverflow.com/questions/35673895/type-hinting-annotation-pep-484-for-numpy-ndarray))